pymc3/variational/opvi.py

```diff
@@ -962,10 +962,23 @@ def _new_initial(self, size, deterministic, more_replacements=None):
         size : scalar
             sample size
         deterministic : bool or scalar
-            whether sample in deterministic manner
+            whether to sample in deterministic manner
         more_replacements : dict
             more replacements passed to shape
 
+        Notes
+        -----
+        Corner case explained
+
+        -   suppose you have AEVB setup
+        -   `X` is purely symbolic
+        -   `X.shape[0]` is needed for `initial` second dim
+        -   for inference you replace `X` with data tensor
+        -   `X.shape[0]` in `initial` remains symbolic and can't be replaced
```

Comment:
Nice. But please format this better, do you mean the following:
```
Suppose you have a AEVB setup that:
- input `X` is purely symbolic, and `X.shape[0]` is needed to `initial` second dim
- to perform inference, `X` is replaced with data tensor, however, since `X.shape[0]` in `initial` remains symbolic and can't be replaced, you get `MissingInputError`
- as a solution, here we perform a manual replacement for the second dim in `initial` here.
```